homeassistant/components/wiz/light.py

```diff
@@ -16,83 +17,177 @@
+        _LOGGER.warning("Bulb is not present in the library. Fallback to full feature")
+        return DEFAULT_COLOR_MODES
+
+
+def supports_effects(bulb_type: BulbType) -> bool:
+    """Check if a bulb supports effects."""
+    with contextlib.suppress(WizLightNotKnownBulb):
+        return bool(bulb_type.features.effect)
+    return True  # default is true
+
+
+def get_min_max_mireds(bulb_type: BulbType) -> tuple[int, int]:
+    """Return the coldest and warmest color_temp that this light supports."""
+    if bulb_type is None:
+        return DEFAULT_MIN_MIREDS, DEFAULT_MAX_MIREDS
+    # DW bulbs have no kelvin
+    if bulb_type.bulb_type == BulbClass.DW:
+        return 0, 0
+    # If bulbtype is TW or RGB then return the kelvin value
+    try:
+        return color_utils.color_temperature_kelvin_to_mired(
+            bulb_type.kelvin_range.max
+        ), color_utils.color_temperature_kelvin_to_mired(bulb_type.kelvin_range.min)
+    except WizLightNotKnownBulb:
+        _LOGGER.debug("Kelvin is not present in the library. Fallback to 6500")
+        return DEFAULT_MIN_MIREDS, DEFAULT_MAX_MIREDS
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
     """Set up the WiZ Platform from config_flow."""
-    # Assign configuration variables.
-    wiz_data = hass.data[DOMAIN][entry.entry_id]
-    wizbulb = WizBulbEntity(wiz_data.bulb, entry.data.get(CONF_NAME), wiz_data.scenes)
-    # Add devices with defined name
-    async_add_entities([wizbulb], update_before_add=True)
-    return True
+    wiz_data: WizData = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities([WizBulbEntity(wiz_data, entry.title)])
 
 
-class WizBulbEntity(LightEntity):
+class WizBulbEntity(CoordinatorEntity, LightEntity):
     """Representation of WiZ Light bulb."""
 
-    def __init__(self, light: wizlight, name, scenes):
+    def __init__(self, wiz_data: WizData, name: str) -> None:
         """Initialize an WiZLight."""
-        self._light = light
-        self._state = None
-        self._brightness = None
+        super().__init__(wiz_data.coordinator)
+        self._light = wiz_data.bulb
+        bulb_type: BulbType = self._light.bulbtype
+        self._attr_unique_id = self._light.mac
         self._attr_name = name
-        self._rgb_color = None
-        self._temperature = None
-        self._hscolor = None
-        self._available = None
-        self._effect = None
-        self._scenes: list[str] = scenes
-        self._bulbtype: BulbType = light.bulbtype
-        self._mac = light.mac
-        self._attr_unique_id = light.mac
-        # new init states
-        self._attr_min_mireds = self.get_min_mireds()
-        self._attr_max_mireds = self.get_max_mireds()
-        self._attr_supported_features = self.get_supported_features()
+        self._attr_effect_list = wiz_data.scenes
+        self._attr_min_mireds, self._attr_max_mireds = get_min_max_mireds(bulb_type)
```

Comment:
Please move this out of the init method or remove the logging in these functions.

Reply:
The logging is going away in another pass

Reply:
https://github.com/home-assistant/core/pull/65819